### PR TITLE
Deep stacks for `E.when` and for handled async operations

### PR DESCRIPTION
**Pull request by erights**
First step towards reviving the deep stacks for tracking asynchronous causality.

### CODE REVIEW

**Comment by erights** [comment]
Ready for review

**Comment by erights** [comment]
Ready for review

Only intra-vat deep stacks. I cannot possibly do inter-vat deep stacks before I leave, which means not until after tc39.

**Comment by erights** [comment]
Too many CI failures so went back to Draft. :(

**Comment by michaelfig** [comment]
> Too many CI failures so went back to Draft. :(

Please tag me again when CI is passing and you want a review.  I'm unsubscribing for now.

**Comment by erights** [comment]
@kriskowal got tests passing. Thanks!
Ready for review.

File: packages/eventual-send/src/track-turns.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+ * The call to `trackTurns` is itself a sending event, that occurs in some call
+ * stack in some turn number at some event number within that turn. Each call
+ * to any of the returned `TurnStartFn`s is a receiving event that begins a new
+ * turn. This sending event caused each of those receiving events.
+ *
+ * @param {TurnStarterFn[]} funcs
+ * @returns {TurnStarterFn[]}
+ */
+export const trackTurns = funcs => {
+  if (!globalThis.assert) {
```
Comment (erights) [inline packages/eventual-send/src/track-turns.js]: To work on earlier JSes, should we also feature test on the existence of `globalThis`? Given the purpose of this shim, probably yes.

Comment (erights) [reply packages/eventual-send/src/track-turns.js]: Done. Resolving.

File: packages/eventual-send/src/track-turns.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+      ((...args) => {
+        hiddenPriorError = sendingError;
+        hiddenCurrentTurn += 1;
+        hiddenCurrentEvent = 0;
+        try {
+          return func(...args);
+        } catch (err) {
+          assert.note(
+            err,
+            assert.details`_thrown from_ ${hiddenPriorError}:${hiddenCurrentTurn}.${hiddenCurrentEvent}`,
```
Comment (kriskowal) [inline packages/eventual-send/src/track-turns.js]: What’s the function of the underbars here? I feel parens or a single colon might be a better compromise for human+computer readability, or no enclosure at all if just for humans. I believe “From prior event” is favored by existing debuggers.

**Review by kriskowal — APPROVED** [review]

File: packages/eventual-send/src/track-turns.js (hunk 2 above)
Comment (erights) [reply packages/eventual-send/src/track-turns.js]: I got rid of the underbars as you suggested, but I didn't change the wording to "prior event". Still ok?

**Comment by kriskowal** [comment]
Very nice

**Pull request merged**